### ISSUE TITLE
Configure template paths for debug and release.

### ISSUE THREAD

**Issue by RylanYancey** [djc/askama#1103]
I'm using Vite for my application, which is used to build the app, but does not run during development. While in debug mode, I want askama to load templates from `/static`, but load from `/dist` during dev. it would be nice if I could configure a debug mode in the Askama.toml without having to do a bunch of ugly #[cfg_attr on each of my templates.

**Comment by djc** [comment]
Hmm, I don't think there is a clean way to do this, but feel free to propose a design.

**Comment by RylanYancey** [comment]
> Hmm, I don't think there is a clean way to do this, but feel free to propose a design.

# Option 1: Set alternative debug path in the macro.
```rs
#[derive(Template)]
#[template(
     path = "./static/pages/settings.html",
     debug = "./dist/pages/settings.html"
)]
pub struct SettingsPage {
    pub profile: UserProfile,
}
```
Askama would use the HTML file in `dist` when building for release, and `static` when building in debug mode. 

# Option 2: Add a 'debug-dirs' key to the config file
```toml
[general]
dirs = ["./static/pages", "./static/templates"]
debug-dirs = ["./dist/pages", "./static/templates"]
```

**Comment by RylanYancey** [comment]
> How would Askama detect whether the calling code is compiled in debug mode? As I understand it, each crate may be compiled with a separate profile, and as a procedural macro crate, the profile with which Askama is compiled doesn't necessarily match the profile the calling code is compiled with.

I was under the impression that `#[cfg(debug_assertions)]` worked for dependencies as well. In the case of option 2, this could be implemented by conditionally using debug-dirs instead of dirs when compiling for debug mode. 

If that is not the case, perhaps this could work:

# Option 3: 'debug' feature in dev-dependencies

```toml
[dependencies]
askama = { version = "0.12.1", features = ["with-axum"] }

[dev-dependencies]
askama = { version = "0.12.1", features = ["with-axum", "debug"]  }
```

Specifying the debug feature in the dev dependencies would cause Askama to use a `debug` profile, such as:

```toml
[general]
dirs = ["./dist"]

[debug]
dirs = ["./static"]
```

**Comment by djc** [comment]
Yeah, I don't think any of this really makes sense. I suggest you write a custom procedural macro to abstract over the `bunch of ugly #[cfg_attr` on your templates.
